tests/integration/targets/mso_schema_site_contract_service_graph/tasks/main.yml

```diff
@@ -0,0 +1,770 @@
+  ignore_errors: true
+  register: nt_add_sg_without_tenant
+
+- name: Assertions check for the negative site contract tests
+  assert:
+    that:
+      - nt_with_invalid_sg_schema is not changed
+      - nt_with_invalid_sg_schema.current == {}
+      - nt_with_invalid_sg_schema.msg == "Provided service_graph_schema{{':'}} 'ansible_schema_temp' does not exist."
+      - nt_add_sg_without_node_relationship is not changed
+      - nt_add_sg_without_node_relationship.current == {}
+      - nt_add_sg_without_node_relationship.msg == "Missing either node_relationship or tenant required attribute."
+      - nt_add_sg_without_tenant is not changed
+      - nt_add_sg_without_tenant.current == {}
+      - nt_add_sg_without_tenant.msg == "Missing either node_relationship or tenant required attribute."
+
+- name: Query a invalid site contract ansible_service_graph_temp
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *mso_info
+    schema: ansible_schema
+    template: ansible_template1
+    site: ansible_test
+    contract: Contract1
+    service_graph: ansible_service_graph_temp
+    state: query
+  ignore_errors: true
+  register: query_sg_with_wrong_name
+
+- name: Query a empty site contract without passing the service graph name
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *mso_info
+    schema: ansible_schema
+    template: ansible_template1
+    site: ansible_test
+    contract: Contract3
+    state: query
+  ignore_errors: true
+  register: query_empty_contract_without_sg_name
+
+- name: Query a invalid site contract without passing the service graph value
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *mso_info
+    schema: ansible_schema
+    template: ansible_template1
+    site: ansible_test
+    contract: Contract4
+    state: query
+  ignore_errors: true
+  register: query_invalid_contract_sg
+
+- name: Query all site contract service graphs with an empty template
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *mso_info
+    schema: ansible_schema
+    template: ansible_template2
+    site: ansible_test
+    service_graph: ansible_service_graph
+    state: query
+  ignore_errors: true
+  register: query_all_sg_with_empty_template
+
+- name: Query a site contract - Contract1 service graph without passing service graph name
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *mso_info
+    schema: ansible_schema
+    template: ansible_template1
+    contract: Contract1
+    site: ansible_test
+    state: query
+  ignore_errors: true
+  register: query_sg_without_name
+
+- name: Query all site contract service graphs without passing the service_graph and contract name
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *mso_info
+    schema: ansible_schema
+    template: ansible_template1
+    site: ansible_test
+    state: query
+  ignore_errors: true
+  register: query_all_sg_without_contract_sg
+
+- name: Assertions check for the query results
```

Comment:
Are we not testing that we get the right amounts returned on all and 1? Also are we not validating the site_contract_service_graph  match?